libs-sol/token-locker/contracts/TokenLocker.sol

```diff
@@ -50,16 +51,18 @@ contract TokenLocker {
     }
 
     function unlockToken(bytes memory proofData, uint64 proofBlockHeight) public {
-        bytes32 key = keccak256(proofData);
-        require(!usedEvents_[key], "The burn event cannot be reused");
-        usedEvents_[key] = true;
         require(prover_.proveOutcome(proofData, proofBlockHeight), "Proof should be valid");
 
         // Unpack the proof and extract the execution outcome.
         Borsh.Data memory borshData = Borsh.from(proofData);
         ProofDecoder.FullOutcomeProof memory fullOutcomeProof = borshData.decodeFullOutcomeProof();
         require(borshData.finished(), "Argument should be exact borsh serialization");
 
+        bytes32 blockHash = fullOutcomeProof.outcome_proof.block_hash;
+        bytes32 receiptId = fullOutcomeProof.outcome_proof.outcome_with_id.outcome.receipt_ids[0];
+        require(!usedEvents_[blockHash][receiptId], "The burn event cannot be reused");
+        usedEvents_[blockHash][receiptId] = true;
```

Comment:
AFAIR receipt ids are globally unique CC @evgenykuzyakov , so it is sufficient to memorize receipt id only.

Reply:
That's great!

> Also, please rebuild the contracts.

Token locker contracts is not commited: https://github.com/near/rainbow-bridge/tree/master/libs-sol/token-locker/dist